src/vkbd/usb-gadget.c

```diff
@@ -0,0 +1,120 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <string.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include "../keys.h"
+#include "usb-gadget.h"
+
+uint16_t mods = 0;
+
+struct vkbd {
+	int fd;
+};
+
+struct hid_report {
+	uint16_t mask;
+	uint16_t code;
+	uint32_t fill;
+};
+
+static int create_virtual_keyboard(void)
+{
+
+	int fd = open("/dev/hidg0", O_WRONLY | O_NONBLOCK);
+	if (fd < 0) {
+		exit(-1);
+	}
+
+	return fd;
+}
+
+struct vkbd *vkbd_init(const char *name)
+{
+	struct vkbd *vkbd = calloc(1, sizeof vkbd);
+	vkbd->fd = create_virtual_keyboard();
+
+	return vkbd;
+}
+
+uint16_t hid_code(uint16_t code)
+{
+	int codes = sizeof(evdev_to_hid)/sizeof(evdev_to_hid[0]);
```

Comment:
Instead of scanning each entry you can create a sparse array literal and just do a lookup

e.g

```
static uint16_t hid_table[] = {
[KEY_A] = 0x4,
[KEY_B] = 0x5,

};

hid_table[code]
```

Reply:
Better indeed. Updated.